src/nevermined/Assets.ts

```diff
@@ -345,6 +428,105 @@ export class Assets extends Instantiable {
+        publishMetadata: PublishMetadata = PublishMetadata.JustMarketplaceAPI,
+        txParams?: TxParameters
+    ): SubscribablePromise<UpdateProgressStep, DDO> {
+
+        this.logger.log('Updating Asset')
+        return new SubscribablePromise(async observer => {
+            observer.next(UpdateProgressStep.ResolveAsset)
+            const ddo = await this.resolve(did)
+            
+
+            observer.next(UpdateProgressStep.UpdateMetadataInDDO)
+            const metadataService = ddo.findServiceByType('metadata')
+            
+            metadataService.attributes.additionalInformation = metadata.additionalInformation
+            metadataService.attributes.main = metadata.main
+            
+            await ddo.replaceService(
+                metadataService.index,
+                metadataService
+            )
+
+            observer.next(UpdateProgressStep.CalculateChecksum)
+            ddo.proof = await ddo.generateProof(publisher.getId())
+            const checksum = ddo.checksum(ddo.shortId())
+
+            observer.next(UpdateProgressStep.AddVersionInDDO)
+            
+            let lastIndex = 0
+            ddo._nvm.versions.map(v => v.id > lastIndex? lastIndex = v.id : false)
+            const ddoVersion: NvmConfigVersions = {
+                id: lastIndex + 1,
+                updated: new Date().toISOString().replace(/\.[0-9]{3}/, ''),
+                checksum: checksum,
+                immutableUrl: ''
+            }
+            ddo._nvm.versions.push(ddoVersion)
+
+            if (publishMetadata != PublishMetadata.JustMarketplaceAPI) {
+                observer.next(UpdateProgressStep.StoringImmutableDDO)
+                ;({url: ddoVersion.immutableUrl, backend: ddoVersion.immutableBackend } = 
+                    await this.publishImmutableContent(ddo, publishMetadata))
+
+                if (ddoVersion.immutableBackend)
+                    ddo._nvm.versions[lastIndex+1] = ddoVersion                    
+            }
+
+            observer.next(UpdateProgressStep.UpdatingAssetOnChain)            
+            await this.nevermined.keeper.didRegistry.updateMetadataUrl(
+                ddo.id,
+                checksum,                
+                publisher.getId(),
+                metadataService.serviceEndpoint,
+                ddoVersion.immutableUrl,
+                txParams
+            )
+
+            observer.next(UpdateProgressStep.StoringDDOMarketplaceAPI)
+            const storedDdo = await this.nevermined.metadata.updateDDO(ddo.id, ddo)
+            
+            observer.next(UpdateProgressStep.AssetUpdated)
+
+            return storedDdo
+        })
+
+    }
+
+
+    private async publishImmutableContent(
+        ddo: DDO,
+        publishMetadata: PublishMetadata = PublishMetadata.IPFS
+    ): Promise<{url: string, backend: ImmutableBackendsSupported}> {
+        let url, backend = undefined
+
+        if (publishMetadata === PublishMetadata.Filecoin) {
```

Comment:
if `uploadContent` fail I think we should catch the error no?

Reply:
see previous comment, when upload fails throws an exception that is managed in the method calling this one